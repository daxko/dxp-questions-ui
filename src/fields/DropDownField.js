var React = require('react');

var DropDownField = React.createClass({

	onChange: function(e) {
		this.props.onChange(e);
		this.props.onBlur();
	},

	render: function() {

		var question = this.props.question;

		return (
			<div className="dxp-dropdown">
				<select onChange={this.onChange} 
					onBlur={this.props.onBlur}
					disabled={this.props.question.read_only}
					className={this.props.className}
					value={this.props.answer || this.props.question.default_value}
					>
					{ !question.hide_empty_option && <option key="default" value="">-Select One-</option> }
					{ 
						question.possible_answers.map(function(pa) {
							return <option key={pa.id} value={pa.id}>{pa.display_value}</option>
						})
					}
				</select>
			</div>
		);
	}

});

module.exports = DropDownField;