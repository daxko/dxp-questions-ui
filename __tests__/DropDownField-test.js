import React from 'react'
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import DropDownField from '../src/fields/DropDownField'
const json = require('../snippets/json/dropdown_question.json')

configure({adapter: new Adapter() })

describe('DropDownField component', () => {
  it('should render a question', () => {
    const wrapper = shallow(<DropDownField question={json}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
