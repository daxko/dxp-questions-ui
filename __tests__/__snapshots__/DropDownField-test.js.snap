// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownField component should render a question 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropDownField
    question={
      Object {
        "default_value": "U",
        "description": "",
        "empty_value": "U",
        "hide_empty_option": true,
        "possible_answers": Array [
          Object {
            "display_value": "Unspecified",
            "id": "U",
            "value": "Unspecified",
          },
          Object {
            "display_value": "Male",
            "id": "M",
            "value": "Male",
          },
          Object {
            "display_value": "Female",
            "id": "F",
            "value": "Female",
          },
        ],
        "read_only": false,
        "required": true,
        "title": "Gender",
        "type": "dropdown",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <select
        className={undefined}
        disabled={false}
        onBlur={undefined}
        onChange={[Function]}
        value="U"
      >
        <option
          value="U"
        >
          Unspecified
        </option>
        <option
          value="M"
        >
          Male
        </option>
        <option
          value="F"
        >
          Female
        </option>
      </select>,
      "className": "dxp-dropdown",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          Array [
            <option
              value="U"
            >
              Unspecified
            </option>,
            <option
              value="M"
            >
              Male
            </option>,
            <option
              value="F"
            >
              Female
            </option>,
          ],
        ],
        "className": undefined,
        "disabled": false,
        "onBlur": undefined,
        "onChange": [Function],
        "value": "U",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": "U",
          "nodeType": "host",
          "props": Object {
            "children": "Unspecified",
            "value": "U",
          },
          "ref": null,
          "rendered": "Unspecified",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "M",
          "nodeType": "host",
          "props": Object {
            "children": "Male",
            "value": "M",
          },
          "ref": null,
          "rendered": "Male",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "F",
          "nodeType": "host",
          "props": Object {
            "children": "Female",
            "value": "F",
          },
          "ref": null,
          "rendered": "Female",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <select
          className={undefined}
          disabled={false}
          onBlur={undefined}
          onChange={[Function]}
          value="U"
        >
          <option
            value="U"
          >
            Unspecified
          </option>
          <option
            value="M"
          >
            Male
          </option>
          <option
            value="F"
          >
            Female
          </option>
        </select>,
        "className": "dxp-dropdown",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            Array [
              <option
                value="U"
              >
                Unspecified
              </option>,
              <option
                value="M"
              >
                Male
              </option>,
              <option
                value="F"
              >
                Female
              </option>,
            ],
          ],
          "className": undefined,
          "disabled": false,
          "onBlur": undefined,
          "onChange": [Function],
          "value": "U",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": "U",
            "nodeType": "host",
            "props": Object {
              "children": "Unspecified",
              "value": "U",
            },
            "ref": null,
            "rendered": "Unspecified",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "M",
            "nodeType": "host",
            "props": Object {
              "children": "Male",
              "value": "M",
            },
            "ref": null,
            "rendered": "Male",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "F",
            "nodeType": "host",
            "props": Object {
              "children": "Female",
              "value": "F",
            },
            "ref": null,
            "rendered": "Female",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
